plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/interact/TooltipSpecFactory.kt

```diff
@@ -9,54 +9,33 @@ import jetbrains.datalore.base.geometry.DoubleVector
 import jetbrains.datalore.base.values.Color
 import jetbrains.datalore.base.values.Pair
 import jetbrains.datalore.plot.base.Aes
-import jetbrains.datalore.plot.base.interact.ContextualMapping
-import jetbrains.datalore.plot.base.interact.GeomTarget
-import jetbrains.datalore.plot.base.interact.MappedDataAccess
+import jetbrains.datalore.plot.base.interact.*
 import jetbrains.datalore.plot.base.interact.MappedDataAccess.MappedData
-import jetbrains.datalore.plot.base.interact.TipLayoutHint
-import jetbrains.datalore.plot.builder.map.GeoPositionField.POINT_X
-import jetbrains.datalore.plot.builder.map.GeoPositionField.POINT_X1
-import jetbrains.datalore.plot.builder.map.GeoPositionField.POINT_Y
-import jetbrains.datalore.plot.builder.map.GeoPositionField.POINT_Y1
 import jetbrains.datalore.plot.builder.presentation.Defaults.Common.Tooltip.AXIS_RADIUS
 import jetbrains.datalore.plot.builder.presentation.Defaults.Common.Tooltip.AXIS_TOOLTIP_COLOR
 
 class TooltipSpecFactory(
-    contextualMapping: ContextualMapping,
-    private val axisOrigin: DoubleVector) {
-
-    private val myTooltipAes: List<Aes<*>> = contextualMapping.tooltipAes
-    private val myAxisAes: List<Aes<*>> = contextualMapping.axisAes
-    private val myDataAccess: MappedDataAccess = contextualMapping.dataAccess
-
+    private val contextualMapping: ContextualMapping,
+    private val axisOrigin: DoubleVector
+) {
     fun create(geomTarget: GeomTarget): List<TooltipSpec> {
         return ArrayList(Helper(geomTarget).tooltipSpecs)
     }
 
     private inner class Helper(private val myGeomTarget: GeomTarget) {
         internal val tooltipSpecs = ArrayList<TooltipSpec>()
-        private val myShortLabels = ArrayList<String>()
-        private val myAesWithoutHint = ArrayList(myTooltipAes)
+        private val myDataAccess: MappedDataAccess = contextualMapping.dataAccessor.mappedDataAccess
+        private val generatedTooltipLines = contextualMapping.generateLines(
```

Comment:
We don't need generic list of lines. Every time we group them into axis, outliler and general tooltip lines.